apollo-gradle-plugin/src/main/java/com/apollographql/android/gradle/ApolloCodeGenInstallTask.java

```diff
@@ -7,46 +7,39 @@
 
 import org.gradle.api.Task;
 import org.gradle.api.specs.Spec;
-import org.gradle.api.tasks.OutputDirectory;
 
-import com.google.common.collect.Lists;
-
-import com.moowork.gradle.node.npm.NpmTask;
-import com.squareup.moshi.JsonReader;
+import com.moowork.gradle.node.npm.NpmInstallTask;
 import com.squareup.moshi.JsonWriter;
 import com.squareup.moshi.JsonAdapter;
 import com.squareup.moshi.Moshi;
 
 import okio.Okio;
 
-public class ApolloCodeGenInstallTask extends NpmTask {
+public class ApolloCodeGenInstallTask extends NpmInstallTask {
   static final String NAME = "installApolloCodegen";
   private static final String INSTALL_DIR = "node_modules/apollo-codegen";
   private static final String APOLLOCODEGEN_VERSION = "0.10.5";
 
-  @OutputDirectory private File installDir;
-
   public ApolloCodeGenInstallTask() {
     // TODO: set to const when ApolloPlugin is in java
     setGroup("apollo");
     setDescription("Runs npm install for apollo-codegen");
-    installDir = getProject().file(INSTALL_DIR);
 
     final File apolloPackageFile = getProject().file("package.json");
     final String apolloVersion = getApolloVersion();
 
     getOutputs().upToDateWhen(new Spec<Task>() {
       public boolean isSatisfiedBy(Task element) {
-        return !(apolloPackageFile.isFile() || (apolloVersion != null && apolloVersion.equals(APOLLOCODEGEN_VERSION)));
+        return apolloVersion != null && apolloVersion.equals(APOLLOCODEGEN_VERSION);
```

Comment:
This check was wrong before. So the task never got outdated.

npm attempts to overwrite to rename and delete the files every time which doesn't make windows happy.